test/FunctionalTests/Microsoft.OData.Client.Tests/Tracking/DataServiceContextQueryTests.cs

```diff
@@ -0,0 +1,189 @@
+                    <ReferentialConstraint Property=""OrgId"" ReferencedProperty=""Id"" />
+                </NavigationProperty>
+            </EntityType>
+            <EntityType Name=""Organization"">
+                <Key>
+                    <PropertyRef Name=""Id"" />
+                </Key>
+                <Property Name=""Id"" Type=""Edm.Int32"" Nullable=""false"" />
+                <Property Name=""Name"" Type=""Edm.String"" Nullable=""false"" />
+            </EntityType>
+        </Schema>
+        <Schema xmlns=""http://docs.oasis-open.org/odata/ns/edm""
+            Namespace=""Sample.API.Models.Enums"">
+            <EnumType Name=""EmployeeType"">
+                <Member Name=""None"" Value=""1"" />
+                <Member Name=""FullTime"" Value=""2"" />
+                <Member Name=""PartTime"" Value=""3"" />
+            </EnumType>
+        </Schema>
+        <Schema xmlns=""http://docs.oasis-open.org/odata/ns/edm"" Namespace=""Default"">
+            <EntityContainer Name=""Container"">
+                <EntitySet Name=""Employees"" EntityType=""Sample.API.Models.Employee"" />
+            </EntityContainer>
+        </Schema>
+    </edmx:DataServices>
+</edmx:Edmx>";
+        #endregion
+
+        public DataServiceContextQueryTests()
+        {
+            var uri = new Uri(ServiceRoot);
+            _defaultContext = new Container(uri);
+        }
+
+        [Fact]
+        public async Task SelectEntities_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 1,
+            ""EmpType"": ""FullTime"",
+            ""OrgId"": 1,
+            ""Name"": ""John Doe""
+        },
+        {
+            ""EmpNumber"": 2,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Jane Doe""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "employees");
+
+            // Act
+            IEnumerable<Employee> employees = await _defaultContext.Employees.ExecuteAsync();
+
+            // Assert
+            Assert.Equal(2, employees.Count());
+        }
+
+        [Fact]
+        public void SelectSpecificEntity_WithEnumAsKey_DoNotThrowException()
+        {
+            // Arrange
+            string response = @"{
+    ""@odata.context"": ""http://localhost:8007/$metadata#Employees"",
+    ""value"": [
+        {
+            ""EmpNumber"": 8,
+            ""EmpType"": ""PartTime"",
+            ""OrgId"": 1,
+            ""Name"": ""Employee Two""
+        }
+    ]
+}";
+            SetupContextWithRequestPipeline(new DataServiceContext[] { _defaultContext }, response, "employees");
+
+            // Act
+            Employee employee = _defaultContext.Employees.Where(e => e.EmpNumber == 8).First();

```

Comment:
Could you have a test where you filter by the enum property (i.e. `e.EmpType`)? And could you have another test where you fetch by key (using `ByKey` with the composite key instead of `Where`)?